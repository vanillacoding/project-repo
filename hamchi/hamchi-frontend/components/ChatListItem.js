import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableWithoutFeedback
} from 'react-native';

import { formatDate } from '../utils';
import colors from '../theme/color';

const ChatListItem = ({ date, user, lastMessage, onPress }) => {
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.container}>
        <View style={styles.leftContainer}>
          <Text style={styles.user}>{user}</Text>
          <Text style={styles.message}>{lastMessage}</Text>
        </View>
        <View style={styles.rightContainer}>
          <Text style={styles.time}>{formatDate(new Date(date))}</Text>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: '95%',
    height: 70,
    justifyContent: 'space-between',
    padding: 10,
    backgroundColor: colors.white,
    marginTop: 10,
    alignSelf: 'center',
    borderRadius: 14,
  },
  leftContainer: {
    flex: 5,
    justifyContent: 'space-around'
  },
  rightContainer: {
    flex: 1,
  },
  user: {
    fontWeight: 'bold',
    fontSize: 16
  },
  message: {
    fontSize: 14,
    color: 'grey',
  },
  time: {
    fontSize: 14,
    color: 'grey'
  }
});

export default ChatListItem;
