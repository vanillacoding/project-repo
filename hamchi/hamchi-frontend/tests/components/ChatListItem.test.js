import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import ChatListItem from '../../components/ChatListItem';

describe('chat list item test', () => {
  it('renders correctly', () => {
    const mockDate = '2021.5.26';
    const mockUser = 'test user';
    const mockLastMessage = 'test last message';

    const tree = renderer.create(
      <ChatListItem
        date={mockDate}
        user={mockUser}
        lastMessage={mockLastMessage}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('show date, user, last message prop', () => {
    const mockEnterChat = jest.fn();
    const mockUser = 'test user';

    const { getByText } = render(
      <ChatListItem
        user={mockUser}
        onPress={mockEnterChat}
      />
    );

    const chatListItem = getByText(mockUser);
    fireEvent.press(chatListItem);
    expect(mockEnterChat).toHaveBeenCalledTimes(1);
  });
});
