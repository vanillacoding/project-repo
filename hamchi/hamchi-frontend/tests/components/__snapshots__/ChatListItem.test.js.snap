// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat list item test renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 14,
      "flexDirection": "row",
      "height": 70,
      "justifyContent": "space-between",
      "marginTop": 10,
      "padding": 10,
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 5,
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      test user
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 14,
        }
      }
    >
      test last message
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 14,
        }
      }
    >
      5월 26일
    </Text>
  </View>
</View>
`;
